### PR TITLE
Update to use new style ruby test reporter

### PR DESCRIPTION
**Pull request by mrb**
Testing changes to https://github.com/codeclimate/ruby-test-reporter/ /c @pbrisbin

### CODE REVIEW

**Comment by wfleming** [comment]
It looks okay to me, and we got coverage for the PR, that all seems great. I'm confused & surprised by the PR coverage status saying coverage dropped 18.2% though?

File: Makefile [hunk 1]
```diff
@@ -13,17 +13,16 @@ test: image
...
+	docker run --entrypoint sh \
+	--env CIRCLECI=$(CIRCLECI) \
+	--env CIRCLE_BUILD_NUM=$(CIRCLE_BUILD_NUM) \
+	--env CIRCLE_BRANCH=$(CIRCLE_BRANCH) \
+	--env CIRCLE_SHA1=$(CIRCLE_SHA1) \
+	--env CODECLIMATE_REPO_TOKEN=$(CODECLIMATE_REPO_TOKEN) \
+	--volume $(PWD)/.git:/usr/src/app/.git:ro \
+	--volume /var/run/docker.sock:/var/run/docker.sock \
+	--volume $(CIRCLE_TEST_REPORTS):/usr/src/app/spec/reports \
+	codeclimate/codeclimate -c "bundle exec rake spec:all spec:benchmark && bundle exec codeclimate-test-reporter"
```
Comment (pbrisbin) [inline Makefile]: 💄 the continuation lines should be indented two spaces below the `docker run`, like they were.

File: Makefile [hunk 2]
```diff
@@ -13,17 +13,16 @@ test: image
...
-	  --env CIRCLE_BUILD_NUM=$(CIRCLE_BUILD_NUM) \
-	  --env CIRCLE_BRANCH=$(CIRCLE_BRANCH) \
-	  --env CIRCLE_SHA1=$(CIRCLE_SHA1) \
-	  --env CODECLIMATE_REPO_TOKEN=$(CODECLIMATE_REPO_TOKEN) \
-	  --entrypoint bundle \
-	  --volume $(PWD)/.git:/usr/src/app/.git:ro \
-	  --volume /var/run/docker.sock:/var/run/docker.sock \
-	  --volume $(CIRCLE_TEST_REPORTS):/usr/src/app/spec/reports \
-	  codeclimate/codeclimate exec rake spec:all spec:benchmark
+	docker run --entrypoint sh \
```
Comment (pbrisbin) [inline Makefile]: 💄 I missed this because it was tucked up here. Can you put it on the next line?

**Comment by mrb** [comment]
@pbrisbin @wfleming Yes almost certainly because of two things:
* Different `SimpleCov` version
* Implicit `SimpleCov` configuration that was happening under the auspices of the old test-reporter, which invoked `SimpleCov`

File: Makefile (hunk 2 above)
Comment (mrb) [inline Makefile]: :+1:

File: Makefile (hunk 1 above)
Comment (mrb) [inline Makefile]: :+1:

**Comment by mrb** [comment]
@pbrisbin @wfleming As per the `citest` target here: https://github.com/codeclimate/codeclimate/blob/update_test_reporter/Makefile#L26 we run:

```
bundle exec rake spec:all spec:benchmark && bundle exec codeclimate-test-reporter
```

Which produces:

```
Coverage report generated for RSpec to /usr/src/app/coverage. 3178 / 3241 LOC (98.06%) covered.
```

And also produces:

```
Coverage report generated for RSpec to /usr/src/app/coverage. 160 / 205 LOC (78.05%) covered.
```

in the build. Reproducing locally, I found that the second results overwrite the first. That's the story there.

**Comment by pbrisbin** [comment]
> Reproducing locally, I found that the second results overwrite the first. That's the story there.

That's what I suspected -- now I'm more curious why the old-style usage wasn't impacted by this. My theory: old-style usage posted both payloads, the first was accepted, the second rejected as a duplicate. New-style usage runs after everything and so only submits the last payload.

If true, one workaround would be to change the command:

```
bundle exec rake spec:all  && bundle exec codeclimate-test-reporter && bundle exec rake spec:benchmark
```

This seems reasonable to me, we want the coverage of the actual test suite, not the benchmarks.

**Comment by mrb** [comment]
@pbrisbin Works for me!

**Comment by mrb** [comment]
@wfleming @pbrisbin Ok after filtering specs and excluding the benchmark results, we're back to the same coverage %.

**Pull request merged**